cloud-api/src/main/java/com/sequenceiq/cloudbreak/cloud/model/LifeCycleState.java

```diff
@@ -0,0 +1,58 @@
+package com.sequenceiq.cloudbreak.cloud.model;
+
+// this is from com.amazonaws.services.elasticfilesystem.model.LifeCycleState
+public enum LifeCycleState {
+
+    PREPARE("prepare"),
+    CREATING("creating"),
+    AVAILABLE("available"),
+    UPDATING("updating"),
+    DELETING("deleting"),
+    DELETED("deleted");
+
+    private String value;
+
+    LifeCycleState(String value) {
+        this.value = value;
+    }
+
+    @Override
+    public String toString() {
+        return this.value;
+    }
+
+    /**
+     * Create instance from input string
+     *
+     * @param value the input in String to be converted to the enum
+     * @return LifeCycleState corresponding state to the input value
+     * @throws IllegalArgumentException If the specified value does not map to one of the known values in this enum.
+     */
+    public static LifeCycleState fromValue(String value) {
+        if (value == null || "".equals(value)) {
+            throw new IllegalArgumentException("Value cannot be null or empty!");
+        }
+
+        for (LifeCycleState enumEntry : LifeCycleState.values()) {
+            if (enumEntry.toString().equals(value)) {
+                return enumEntry;
+            }
+        }
+
+        throw new IllegalArgumentException("Cannot create enum from " + value + " value!");
```

Comment:
You can replace pretty all the functionality of this method with:
```
if (Strings.isNullOrEmpty(value)) {
    throw new IllegalArgumentException("Value cannot be null or empty!");
}
return LifeCycleState.valueOf(value.toUpperCase());
```

Reply:
Thanks! it should be LifeCycleState.valueOf(value.toLowerCase()).

Reply:
Your enums are in upper case though, right? `valueOf` converts a string to an enum, if the string matches the enum name. It's not related to the "value" you're manually assigning to the enums. In fact, I don't think you need to explicitly assign a "value" at all.